firmware/firmware.md

```diff
@@ -11262,12 +11262,16 @@ make: *** [user] Error 2
 
 When you are using the Particle Cloud to compile your `.ino` source code, a preprocessor comes in to modify the code into C++ requirements before producing the binary file used to flash onto your devices.
 
-However, there might be instances where the preprocessor causes issues in your code. One example is the use of class/structs in your function parameters.
+However, there might be instances where the preprocessor causes issues in your code. Examples include:
 
+  * Using class/struct types in your function parameters (as above).
+  * Using single variables in conditions - e.g. `else if(foo)`
```

Comment:
Agreed - I'd rather the preprocessor was fixed than documenting around it. However, preprocessor issues (including this one) have been discussed on the forums for a couple of years now, so I figured doc updates had a better chance of succeeding.

Additionally, there are mixed messages in the forum about including `Particle.h` or `application.h`, so I figured this should also be documented.